fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/AbstractSumoAmbassador.java

```diff
@@ -1193,7 +1196,7 @@ public synchronized void processTimeAdvanceGrant(long time) throws InternalFeder
         }
 
         // schedule events, e.g. change speed events
-        int scheduled = eventScheduler.scheduleEvents(time);
+        int scheduled = eventScheduler.scheduleEvents(convertTime((int) time));
```

Comment:
I think this is wrong. Why should the time here be converted? `time` is already in nanoseconds and should stay nanoseconds